checker/src/main/java/org/checkerframework/checker/optional/OptionalChecker.java

```diff
@@ -14,6 +15,7 @@
 // @NonNull, make the return type have type @Present.
 @RelevantJavaTypes(Optional.class)
 @StubFiles({"javaparser.astub"})
+@SupportedOptions("optionalMapAssumeNonNull")
```

Comment:
I think a better name would be "assumeNullAnnotated" or the like.  This isn't specific to Optional.map, and it's about the assumption that nullness annotations are available for method references.

Reply:
The code as implemented is specific to Optional.map.  (And is specific to method references.)